### PR TITLE
Better documentation for buzz user function

### PR DESCRIPTION
**Pull request by xgroleau**

### CODE REVIEW

File: src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h [hunk 1]
```diff
@@ -21,39 +21,49 @@ namespace BittyBuzzUserFunctions {
...
-     * The table description of the arguments is an array of tuple, the tuple has the arg name as
-     *key and the type, you can use any value, only the type is checked. You could define a variable
-     *as int=0 float=0.0 And use those variables for types description
+     *@details This closure expects four parameters, the closure can be a lambda.
+     * -# stringId (name of the function)
+     * -# a closure (the callback function)
+     * -# a table with the description of the arguments
+     * -# one with the self context
+     *
+     * The table description of the arguments is an array of tuple, the
```
Comment (t-constantin) [inline src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h:30]: tuple"s"

Comment (t-constantin) [reply src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h:30]: "... is an array of tuples, which hold the arg name as key and a value that can be of any type. Only the type is checked."
would probably be better

File: src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h [hunk 2]
```diff
@@ -21,39 +21,49 @@ namespace BittyBuzzUserFunctions {
...
+     *
      * function registered_function(arg_int, arg_float) {
-     *   assert_true(arg_int == 42);
-     *    assert_true(arg_float == 42.24);
+     *   log("Context: ", self.some_context, "Arg int: ", arg_int, "Arg float: ", arg_float);
      * }
      * var args_description = {
-     *    .0 = {.arg_int=0},
-     *    .1 = {.arg_float=0.0}
+     *    .0 = {.arg_int=0}, // We juste pass value so the type is registered
```
Comment (t-constantin) [inline src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h:41]: typo: just

File: src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h [hunk 3]
```diff
@@ -21,39 +21,49 @@ namespace BittyBuzzUserFunctions {
...
     /**
      *@brief calls a function to a host
-     *@details This closure expects tree parameters. The is of the host to call (0 for prodcast, id
-     *for local host), the name of the function, a table with the list of arguments
+     *@details This closure expects tree parameters.
+     * -# the id of the host to call (0 for broadcast, id for local host)
+     * -# the name of the function
+     * -# a table with the list of arguments
+     *
+     * The table argument must be by index from 0 to N-1 args.
```
Comment (t-constantin) [inline src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h:55]: typo: argument"s"

**Review by t-constantin — CHANGES_REQUESTED** [review]
Small typo corrections needed

File: src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h [hunk 4]
```diff
@@ -21,39 +21,49 @@ namespace BittyBuzzUserFunctions {
...
-     * register_closure("registeredFunction", registered_function, args_description, nil)
+     * register_closure("registeredFunction", registered_function, args_description, context);
      *@endcode */
     void registerClosure();
 
     /**
      *@brief calls a function to a host
-     *@details This closure expects tree parameters. The is of the host to call (0 for prodcast, id
-     *for local host), the name of the function, a table with the list of arguments
+     *@details This closure expects tree parameters.
```
Comment (stlaperriere) [inline src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h:50]: Do you mean "three" (3) parameters?

**Review by stlaperriere — CHANGES_REQUESTED** [review]

File: src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h (hunk 4 above)
Comment (xgroleau) [reply src/bittybuzz/include/bittybuzz/BittyBuzzUserFunctions.h:50]: :deciduous_tree:  :deciduous_tree:

**Review by stlaperriere — APPROVED** [review]

**Review by t-constantin — APPROVED** [review]

**Pull request merged**